src/Microsoft.AspNetCore.Sockets.Client/Connection.cs

```diff
@@ -194,20 +193,21 @@ public Task<bool> SendAsync(byte[] data, MessageType type)
 
             if (_connectionState != ConnectionState.Connected)
             {
-                return false;
+                throw new InvalidOperationException(
+                    $"Cannot send messages when the connection is not in the Connected state.");
             }
 
-            var message = new Message(data, type);
+            var sendTcs = new TaskCompletionSource<object>();
+            var message = new SendMessage(data, type, sendTcs);
 
             while (await Output.WaitToWriteAsync(cancellationToken))
             {
                 if (Output.TryWrite(message))
                 {
-                    return true;
+                    await sendTcs.Task;
```

Comment:
Is it OK for the code following the call Connection.SendAsync() to run inline with Transport.SendMessages? I'm worried about callers blocking.